modules/javafx.controls/src/main/java/javafx/scene/control/Control.java

```diff
@@ -222,8 +222,14 @@ private static Class<?> loadClass(final String className, final Object instance)
      * {@code Skin}. Every {@code Skin} maintains a back reference to the
      * {@code Control} via the {@link Skin#getSkinnable()} method.
      * <p>
+     * To ensure a one-to-one relationship between a {@code Control} and its {@code Skin},
+     * {@link Control#setSkin(Skin)} will check the return value of {@link Skin#getSkinnable()}
+     * against this Control, and throw an {@code IllegalArgumentException} if it is not the same.
+     * <p>
      * A skin may be null.
```

Comment:
This breaks the 1-1 relationship mentioned above, so it's probably best to mention this as an exception to the 1-1 rule.

Reply:
Maybe something like `will check the return value of {@link Skin#getSkinnable()}, if the Skin is not {@code null}, against this Control...`?

Reply:
How about

```
     * To ensure a one-to-one relationship between a {@code Skinnable} and its
     * {@code Skin}, some implementations of {@link Skinnable#setSkin(Skin)} method will check
     * the return value of {@link Skin#getSkinnable()}, and if it is not {@code null}, compare it
     * with this Skinnable, throwing an {@code IllegalArgumentException} if it is not the same.
```